keepalived/check/check_parser.c
=======================================================
lhs: 100644 | 382861c421554369b8c844be74a0c335715669dd
rhs: 100644 | 8e86be0db19caa9353ae2eb6e7b4ce32377de84b
---@@ -329,6 +329,10 @@ notify_up_handler(vector_t *strvec)
 {
 	virtual_server_t *vs = LIST_TAIL_DATA(check_data->vs);
 	real_server_t *rs = LIST_TAIL_DATA(vs->rs);
+	if (rs->notify_up) {
+		log_message(LOG_INFO, "(%s): notify_up script already specified - ignoring %s", vs->vsgname, FMT_STR_VSLOT(strvec,1));
+		return;
+	}
 	rs->notify_up = set_check_notify_script(strvec);
 }
 static void
@@ -336,6 +340,10 @@ notify_down_handler(vector_t *strvec)
 {
 	virtual_server_t *vs = LIST_TAIL_DATA(check_data->vs);
 	real_server_t *rs = LIST_TAIL_DATA(vs->rs);
+	if (rs->notify_down) {
+		log_message(LOG_INFO, "(%s): notify_down script already specified - ignoring %s", vs->vsgname, FMT_STR_VSLOT(strvec,1));
+		return;
+	}
 	rs->notify_down = set_check_notify_script(strvec);
 }
 static void

---